doc/guide/transactions-and-threading.rst

```diff
@@ -0,0 +1,394 @@
+  application using the transactional system doesn't have to undo
+  partial changes.  This takes a significant burden from developers
+  and increases the reliability of applications.
+
+Concurrency
+  Transactions provide a way of managing concurrent updates to data.
+  Different programs operate on the data independently, without having
+  to use low-level techniques to moderate their access. Coordination
+  and synchronization happens via transactions.
+
+
+.. _using-transactions-label:
+
+Using transactions
+==================
+
+All activity in ZODB happens in the context of database connections
+and transactions.  Here's a simple example::
+
+  import ZODB, transaction
+  db = ZODB.DB(None) # Use a mapping storage
+  conn = db.open()
+
+  conn.root.x = 1
+  transaction.commit()
+
+.. -> src
+
+   >>> exec(src)
+
+In the example above, we used ``transaction.commit()`` to commit a
+transaction, making the change to ``conn.root`` permanent.  This is
+the most common way to use ZODB, at least historically.
+
+If we decide we don't want to commit a transaction, we can use
+``abort``::
+
+  conn.root.x = 2
+  transaction.abort()
+
+.. -> src
+
+   >>> exec(src)
+   >>> conn.root.x
+   1
+   >>> conn.close()
+
+In this example, because we aborted the transaction, the value of
+``conn.root.x`` was rolled back to 1.
+
+There are a number of things going on here that deserve some
+explanation.  When using transactions, there are three kinds of
+objects involved:
+
+Transaction
+   Transactions represent units of work.  They have beginnings and
+   ends. They provide the
+   :interface:`~transaction.interfaces.ITransaction` interface.
+
+Transaction manager
+   Transaction managers create transactions and
+   provide APIs to start and end transactions.  The transactions
+   managed are always sequential. There is always exactly one active
+   transaction associated with a transaction manager at any point in
+   time. Transaction managers provide the
+   :interface:`~transaction.interfaces.ITransactionManager` interface.
+
+Data manager
+   Data managers manage data associated with transactions.  ZODB
+   connections are data managers.  The details of how they interact
+   with transactions isn't important here.
+
+Explicit transaction managers
+-----------------------------
+
+ZODB connections have transaction managers associated with them when
+they're opened. When we call the database :meth:`~ZODB.DB.open` method
+without an argument, a thread-local transaction manager is used. Each
+thread has it's own transaction manager.  When we called
+``transaction.commit()`` above we were calling commit on the
+thread-local transaction manager.
+
+Because we used a thread-local transaction manager, all of the work in
+the transaction needs to happen in the same thread.  Similarly, only
+one transaction can be active in a thread.
+
+If we want to have transactions who's work is spread over multiple
```

Comment:
who's -> whose

Although I would phrase it as: If we want to spread the work of a transaction over multiple threads,...